packages/core/src/combinator/combine.ts

```diff
@@ -27,16 +27,16 @@ export const combine = <A, B, C>(f: (a: A, b: B) => C, stream1: Stream<A>, strea
 * @returns stream containing the result of applying f to the most recent
 *  event of each input stream, whenever a new event arrives on any stream.
 */
-export const combineArray = <Args extends unknown[], R>(f: (...args: Args) => R, streams: ToStreamsArray<Args>): Stream<R> =>
+export const combineArray = <Args extends unknown[], R>(f: (...args: Args) => R, streams: InputStreamArray<Args>): Stream<R> =>
   streams.length === 0 || containsCanonicalEmpty(streams) ? empty()
     : streams.length === 1 ? map(f as any, streams[0])
-      : new Combine(f, streams)
+      : new Combine(f, streams as Stream<Args>[])
```

Comment:
Hmm!  I am not so happy about this casting because it means we have misaligned types.

Reply:
hey @Frikki , i got rid of the casting, there are still room for improvements in both versions(PR and master), both are not strictly correct, perhaps we can improvement it when typescript gets upgraded in the future.